qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -368,6 +601,11 @@ def einsum(
             )
             # results will be contained in transformed["+-"] and transformed["++--"], respectively
 
+        .. note::
+
+           :class:`sparse.SparseArray` does not support ``numpy.einsum``. Thus, all sparse matrices
+           involved in this operation will be converted to dense matrices!
```

Comment:
> Thus, all sparse matrices involved in this operation will be converted to dense matrices!

Or said another way the resultant PolynomialTensor will be all dense matrices. So if a user wanted sparse as result they would have to call tosparse afterwards. Converting the matrix back to sparse might be possible here I guess on the fly.